webapp/src/ts/services/startup-modals.service.ts

```diff
@@ -0,0 +1,202 @@
+import { Injectable } from '@angular/core';
+
+import { TourService } from './tour.service';
+import { UpdateUserService } from './update-user.service';
+import { SessionService } from './session.service';
+import { TourSelectComponent } from '@mm-modals/tour/tour-select.component';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { SettingsService } from './settings.service';
+import { UserSettingsService } from './user-settings.service';
+
+class StartupModal {
+  required: (settings, user?) => boolean
+  render: () => Promise<any>
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StartupModalsService {
+
+  tours: object[];
+  modalsToShow: StartupModal[];
+  startupModals: StartupModal[] = [
+    // welcome screen
+    {
+      required: settings => !settings.setup_complete,
+      render: () => {
+        //TODO
+        // return Modal({
+        //   templateUrl: 'templates/modals/welcome.html',
+        //   controller: 'WelcomeModalCtrl',
+        //   controllerAs: 'welcomeModalCtrl',
+        //   size: 'lg',
+        // }).catch(() => {});
+        return Promise.resolve();
+      },
+    },
+    // guided setup
+    {
+      required: settings => !settings.setup_complete,
+      render: () => {
+        //TODO
+        // return ctrl.openGuidedSetup()
+        //   .then(() => UpdateSettings({ setup_complete: true }))
+        //   .catch(err => $log.error('Error marking setup_complete', err));
+        return Promise.resolve();
+      },
+    },
+    // tour
+    {
+      required: (settings, user) => !user.known && this.tours.length > 0,
+      render: () => {
+        this.openTourSelect();
+        return this.updateUserService.update(
```

Comment:
Yes the modal is opened without returning any promise because it called passing as argument the component class, not an instance, that is way in the function the execution is not suspended waiting for the return from the modal, but I see that the [ModalService.show()](https://github.com/medic/cht-core/blob/07ad95a5e4641afaa52310780c9162e2d297f563/webapp/src/ts/modals/mm-modal/mm-modal.ts#L81) method can be called passing a handler that is subscribed to be called when the modal is closed, right? I'll take a look...